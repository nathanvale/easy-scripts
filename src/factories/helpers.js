function createDoItForYouPrompt({type = 'select', name, message} = {}) {
  return {
    type,
    name,
    message,
    choices: [
      {
        title: `"No thank you. I can do that myself."`,
        value: false,
      },
      {
        title: `"Yes please! That would be awesome! :)"`,
        value: true,
      },
    ],
    initial: 1,
  }
}

module.exports = {
  createDoItForYouPrompt,
}
