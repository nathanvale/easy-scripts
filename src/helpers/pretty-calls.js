function prettyCalls(calls) {
  const argumentMapper = (item, argIndex) => `Argument ${argIndex + 1}:
    ${item}`
  const reducer = (previousValue, currentValue, currentCallIndex) => `${
    previousValue === '' ? previousValue : `${previousValue}\n`
  }Call ${currentCallIndex + 1}:
  ${currentValue.map(argumentMapper)}`
  return calls.reduce(reducer, '')
}

module.exports = {prettyCalls}
