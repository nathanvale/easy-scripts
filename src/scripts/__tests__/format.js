import cases from 'jest-in-case'
import {unquoteSerializer, winPathSerializer} from '../../helpers/serializers'

jest.mock('../../utils')

expect.addSnapshotSerializer(unquoteSerializer)
expect.addSnapshotSerializer(winPathSerializer)

let fromConfigsMock

cases(
  'format',
  ({args}) => {
    // beforeEach
    jest.resetModules()
    fromConfigsMock = require('../../utils').fromConfigs
    fromConfigsMock.mockReturnValue('path/to/some/config.js')
    const {sync: crossSpawnSyncMock} = require('cross-spawn')
    const originalExit = process.exit
    const originalArgv = process.argv
    const utils = require('../../utils')
    utils.resolveBin = (modName, {executable = modName} = {}) => executable
    process.exit = jest.fn()

    // tests
    process.argv = ['node', '../format', ...args]
    require('../format')
    expect(crossSpawnSyncMock).toHaveBeenCalledTimes(1)
    const [firstCall] = crossSpawnSyncMock.mock.calls
    const [script, calledArgs] = firstCall
    expect([script, ...calledArgs].join(' ')).toMatchSnapshot()

    // afterEach
    process.exit = originalExit
    process.argv = originalArgv
  },
  {
    'calls prettier CLI with args': {
      args: ['my-src/**/*.js'],
    },
    '--no-write prevents --write argument from being added': {
      args: ['--no-write'],
    },
    '--config arg can be used for a custom config': {
      args: ['--config', './my-config.js'],
    },
    '--ignore-path arg can be used for a custom ignore file': {
      args: ['--ignore-path', './.myignore'],
    },
  },
)
