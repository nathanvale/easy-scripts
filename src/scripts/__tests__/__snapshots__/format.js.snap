// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`scripts format --config arg can be used for a custom config 1`] = `prettier --ignore-path path/to/some/config.js --write **/*.+(js|jsx|json|less|css|ts|tsx|md) --config ./my-config.js`;

exports[`scripts format --ignore-path arg can be used for a custom ignore file 1`] = `prettier --config path/to/some/config.js --write **/*.+(js|jsx|json|less|css|ts|tsx|md) --ignore-path ./.myignore`;

exports[`scripts format --no-write prevents --write argument from being added 1`] = `prettier --config path/to/some/config.js --ignore-path path/to/some/config.js **/*.+(js|jsx|json|less|css|ts|tsx|md) --no-write`;

exports[`scripts format calls prettier CLI with args 1`] = `prettier --config path/to/some/config.js --ignore-path path/to/some/config.js --write my-src/**/*.js`;
